packages/react-ai/src/components/AIConversation/views/Controls/FormControl.tsx

```diff
@@ -165,8 +172,23 @@ export const FormControl: FormControl = () => {
   const ref = React.useRef<HTMLFormElement | null>(null);
   const controls = React.useContext(ControlsContext);
   const [composing, setComposing] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
 
   const submitMessage = async () => {
+    const hasInput =
+      !!input?.text || (input?.files?.length && input?.files?.length > 0);
```

Comment:
Can split out independent variables here and reuse for the conditionals on lines 194 and 201, e.g.:
```ts
const hasInputText = !!input?.text;
// an empty array will resolve false when evaluating the length
const hasInputFiles = !!input?.files?.length;

const submitMessage = async () => {
  const hasInput = hasInputText || hasInputFiles;
  // ....
```